File: cpp/06-io-operations/file_io.cpp
#include <iostream>
#include <fstream>
#include <string>

int main() {
    // Writing to a file with an output file stream
    std::ofstream outFile("notes.txt");
    if (!outFile) {
        std::cerr << "Failed to open file for writing\n";
        return 1;
    }
    outFile << "Line 1: C++ file I/O\n";
    outFile << "Line 2: Using fstream\n";
    outFile << "Line 3: RAII closes the file\n";
    outFile.close();

    std::cout << "File written successfully.\n";

    // Reading the file back line by line
    std::ifstream inFile("notes.txt");
    if (!inFile) {
        std::cerr << "Failed to open file for reading\n";
        return 1;
    }

    std::string line;
    int lineNumber = 1;
    while (std::getline(inFile, line)) {
        std::cout << lineNumber << ": " << line << "\n";
        lineNumber++;
    }

    return 0;
}

File: matlab/06-io-operations/file_io.m
% Writing to and reading from a text file

% --- Writing ---
fid = fopen('greetings.txt', 'w');       % open for writing (truncates)
fprintf(fid, 'Hello from MATLAB\n');
fprintf(fid, 'Line number %d\n', 2);
fprintf(fid, 'Pi is approximately %.4f\n', pi);
fclose(fid);                             % always close the handle

disp('File written. Reading it back line by line:');

% --- Reading line by line ---
fid = fopen('greetings.txt', 'r');       % open for reading
line = fgetl(fid);                       % read the first line
while ischar(line)                       % fgetl returns -1 at end of file
    fprintf('  %s\n', line);
    line = fgetl(fid);                   % advance to the next line
end
fclose(fid);

% --- Reading the whole file at once ---
contents = fileread('greetings.txt');
fprintf('Total characters read: %d\n', length(contents));
